libraries/package-extractor/src/scripts/createLinks/cli/actions/CreateLinksAction.ts

```diff
@@ -0,0 +1,128 @@
+import type { ITerminal } from '@rushstack/terminal';
+import { CommandLineAction, type CommandLineFlagParameter } from '@rushstack/ts-command-line';
+import type { IExtractorMetadataJson, IProjectInfoJson } from '../../../../PackageExtractor';
+import { makeBinLinksAsync } from '../../../../Utils';
+import { getExtractorMetadataAsync } from '../../utilities/CreateLinksUtilities';
+import {
+  TARGET_ROOT_FOLDER,
+  REALIZE_FILES_PARAMETER_NAME,
+  LINK_BINS_PARAMETER_NAME,
+  MAX_CONCURRENCY
+} from '../../utilities/constants';
+import { removeLinksAsync } from './RemoveLinksAction';
+
+async function createLinksAsync(
+  terminal: ITerminal,
+  targetRootFolder: string,
+  extractorMetadataObject: IExtractorMetadataJson
+): Promise<void> {
+  await Async.forEachAsync(
+    extractorMetadataObject.links,
+    async (linkInfo) => {
+      // Link to the relative path for symlinks
+      const newLinkPath: string = path.join(targetRootFolder, linkInfo.linkPath);
+      const linkTargetPath: string = path.join(targetRootFolder, linkInfo.targetPath);
+
+      // Make sure the containing folder exists
+      await FileSystem.ensureFolderAsync(path.dirname(newLinkPath));
+
+      // NOTE: This logic is based on NpmLinkManager._createSymlink()
+      if (linkInfo.kind === 'folderLink') {
+        terminal.writeVerboseLine(`Creating linked folder at path "${newLinkPath}"`);
+        await FileSystem.createSymbolicLinkJunctionAsync({ newLinkPath, linkTargetPath });
+      } else if (linkInfo.kind === 'fileLink') {
+        // Use hardlinks for Windows and symlinks for other platforms since creating a symbolic link
+        // requires administrator permission on Windows. This may cause unexpected behaviour for consumers
+        // of the hardlinked files. If this becomes an issue, we may need to revisit this.
+        terminal.writeVerboseLine(`Creating linked file at path "${newLinkPath}"`);
+        if (process.platform === 'win32') {
+          await FileSystem.createHardLinkAsync({ newLinkPath, linkTargetPath });
+        } else {
+          await FileSystem.createSymbolicLinkFileAsync({ newLinkPath, linkTargetPath });
+        }
+      }
+    },
+    { concurrency: MAX_CONCURRENCY }
+  );
+}
+
+async function realizeFilesAsync(
+  terminal: ITerminal,
+  targetRootFolder: string,
+  extractorMetadataObject: IExtractorMetadataJson
+): Promise<void> {
+  await Async.forEachAsync(
+    extractorMetadataObject.files,
+    async (relativeFilePath) => {
+      const filePath: string = `${targetRootFolder}/${relativeFilePath}`;
+      const realFilePath: string = await FileSystem.getRealPathAsync(filePath);
+      if (!Path.isEqual(realFilePath, filePath)) {
+        await FileSystem.deleteFileAsync(filePath);
+
+        // Hard links seem to cause build failures on Mac, so for all other operating
+        // systems we copy files.
+        terminal.writeVerboseLine(`Realizing file at path "${filePath}"`);
+        if (process.platform === 'win32') {
```

Comment:
Linux likes hardlinks just fine. Shouldn't we do the conditions the other order?